vignettes/adsl.Rmd

```diff
@@ -0,0 +1,258 @@
+    condition = (EXDOSE > 0 | (EXDOSE == 0 & str_detect(EXTRT, "PLACEBO")))
+  )
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adsl,
+  display_vars = exprs(USUBJID, TRTSDT, ARM, ACTARM, SAFFL)
+)
+```
+
+## Derive vaccination date Variables  {#vax_date}
+
+In this step, we will create a vaccination date variables from `EX` domain. The function `derive_vars_vaxdt` returns the variables `VAX01DT`,`VAX02DT`... added to the `adsl` dataset based on number of vaccinations.
+
+If there are multiple vaccinations for a visit per subject,warning will be
+provided and only first observation will be filtered based on the variable
+order specified on the `order` argument. In this case, user need to select
+the `by_vars` appropriately.
+
+```{r eval=TRUE}
+adsl <- derive_vars_vaxdt(
+  dataset = ex,
+  dataset_adsl = adsl,
+  by_vars = exprs(USUBJID, VISITNUM),
+  order = exprs(USUBJID, VISITNUM, VISIT, EXSTDTC)
+)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adsl,
+  display_vars = exprs(USUBJID, VAX01DT, VAX02DT)
+)
+```
+
+## Derive Other Variables  {#other}
+The users can add specific code to cover their need for the analysis.
+
+The following functions are helpful for many ADSL derivations:
+
+  - `derive_vars_merged()` - Merge Variables from a Dataset to the Input Dataset
+  - `derive_var_merged_exist_flag()` - Merge an Existence Flag
+  - `derive_var_merged_cat()` - Merge a Categorization Variable
+  - `derive_var_merged_character()` - Merge a Character Variable
+  - `derive_var_merged_summary()` - Merge a Summary Variable
+
+See also [Generic Functions](generic.html).
+
+## Add Labels and Attributes {#attributes}
+
+Adding labels and attributes for SAS transport files is supported by the
+following packages:
+
+- [metacore](https://atorus-research.github.io/metacore/): establish a common
+foundation for the use of metadata within an R session.
+
+- [metatools](https://pharmaverse.github.io/metatools/): enable the use of
+metacore objects. Metatools can be used to build datasets or enhance columns in
+existing datasets as well as checking datasets against the metadata.
+
+- [xportr](https://atorus-research.github.io/xportr/): functionality to
+associate all metadata information to a local R data frame, perform data set
+level validation checks and convert into a [transport v5
+file(xpt)](https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.5/movefile/n1xbwdre0giahfn11c99yjkpi2yb.htm).
+
+NOTE: All these packages are in the experimental phase, but the vision is to
+have them associated with an End to End pipeline under the umbrella of the
+[pharmaverse](https://github.com/pharmaverse). An example of applying metadata 
+and perform associated checks can be found at the [pharmaverse E2E example](https://examples.pharmaverse.org/data/adsl/).
+
+# Example Script 
+
+ADaM | Sample Code
+---- | --------------
+ADSL | [ad_adsl.R](https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adsl.R){target="_blank"} 
```

Comment:
this will need to point the admiralvaccine template eventually

Reply:
We should add the link once it is pushed to main branch right?, we need to rename the template as well. We will update the link later once we are good?

Reply:
that is correct - can you make an issue so we don't forgot to make this update.

Reply:
Sure Ben. Thanks!!